examples/nodejs/cache/doc-example-files/doc-examples-js-apis.ts

```diff
@@ -781,13 +786,93 @@ function example_API_InstantiateTopicClient() {
+    console.log(`Auth token starts with: ${singleCacheROTokenResponse.authToken.substring(0, 10)}`);
+    console.log(`Refresh token starts with: ${singleCacheROTokenResponse.refreshToken.substring(0, 10)}`);
+    console.log(`Expires At: ${singleCacheROTokenResponse.expiresAt.epoch()}`);
+  }
+
+  // Generate a token that can call all data plane APIs on all caches. No topic apis (publish/subscribe) are allowed.
+  const allCachesRWTokenResponse = await authClient.generateAuthToken(
+    TokenScopes.cacheReadWrite(AllCaches),
+    ExpiresIn.minutes(30)
+  );
+  if (allCachesRWTokenResponse instanceof GenerateAuthToken.Success) {
+    console.log('Generated an auth token with read-write access to all caches!');
+    // logging only a substring of the tokens, because logging security credentials is not advisable :)
+    console.log(`Auth token starts with: ${allCachesRWTokenResponse.authToken.substring(0, 10)}`);
+    console.log(`Refresh token starts with: ${allCachesRWTokenResponse.refreshToken.substring(0, 10)}`);
+    console.log(`Expires At: ${allCachesRWTokenResponse.expiresAt.epoch()}`);
+  }
+
+  // Generate a token that can call publish and subscribe on all topics within cache bar
+  const singleCacheAllTopicsRWTokenResponse = await authClient.generateAuthToken(
+    TokenScopes.topicPublishSubscribe({name: 'bar'}, AllTopics),
+    ExpiresIn.minutes(30)
+  );
+  if (singleCacheAllTopicsRWTokenResponse instanceof GenerateAuthToken.Success) {
+    console.log('Generated an auth token with publish-subscribe access to all topics within cache bar!');
+    // logging only a substring of the tokens, because logging security credentials is not advisable :)
+    console.log(`Auth token starts with: ${singleCacheAllTopicsRWTokenResponse.authToken.substring(0, 10)}`);
+    console.log(`Refresh token starts with: ${singleCacheAllTopicsRWTokenResponse.refreshToken.substring(0, 10)}`);
+    console.log(`Expires At: ${singleCacheAllTopicsRWTokenResponse.expiresAt.epoch()}`);
+  }
+
+  // Generate a token that can only call subscribe on topic where_is_mo within cache mo_nuts
+  const oneCacheOneTopicRWTokenResponse = await authClient.generateAuthToken(
+    TokenScopes.topicSubscribeOnly('mo_nuts', 'where_is_mo'),
+    ExpiresIn.minutes(30)
+  );
+  if (oneCacheOneTopicRWTokenResponse instanceof GenerateAuthToken.Success) {
+    console.log('Generated an auth token with subscribe-only access to topic where_is_mo within cache mo_nuts!');
+    // logging only a substring of the tokens, because logging security credentials is not advisable :)
+    console.log(`Auth token starts with: ${oneCacheOneTopicRWTokenResponse.authToken.substring(0, 10)}`);
+    console.log(`Refresh token starts with: ${oneCacheOneTopicRWTokenResponse.refreshToken.substring(0, 10)}`);
+    console.log(`Expires At: ${oneCacheOneTopicRWTokenResponse.expiresAt.epoch()}`);
+  }
+
+  // Generate a token with multiple permissions
+  const permissions = {
```

Comment:
Is it possible to split this out to be two different variables? One for TopicsPermissions and one for CachePermissions? and add more comments like here https://github.com/momentohq/public-dev-docs/blob/chore/new-tokens-docs/docs/develop/api-reference/auth-tokens.md#permission-object-examples

Reply:
Yep, my bad. You're correct. I have it labeled wrong.

What I am trying to show are examples of all 4 of the managed roles and the use of either AllCaches or strings passed in. So it is brutally obvious what the possibilities are.

Reply:
I pushed a version labelled correctly.